framework/src/main/java/org/fulib/fx/controller/internal/ReflectionSidecar.java

```diff
@@ -460,10 +460,12 @@ private boolean keyEventMatchesAnnotation(KeyEvent event, OnKey annotation) {
         return (annotation.code() == KeyCode.UNDEFINED || event.getCode() == annotation.code()) &&
             (annotation.character().isEmpty() || event.getCharacter().equals(annotation.character())) &&
             (annotation.text().isEmpty() || event.getText().equals(annotation.text())) &&
-            (event.isShiftDown() || !annotation.shift()) &&
-            (event.isControlDown() || !annotation.control()) &&
-            (event.isAltDown() || !annotation.alt()) &&
-            (event.isMetaDown() || !annotation.meta());
+
+            // key pressed ==> not strict || key required
+            (event.isShiftDown() && (!annotation.strict() || annotation.shift())) &&
+            (event.isControlDown() && (!annotation.strict() || annotation.control())) &&
+            (event.isAltDown() && (!annotation.strict() || annotation.alt())) &&
+            (event.isMetaDown() && (!annotation.strict() || annotation.meta()));
```

Comment:
This boolean logic seems wrong, if the key is not down, the condition is never true

Reply:
Fixed in latest commit